src/converse-muc-views.js

```diff
@@ -797,6 +797,13 @@ converse.plugins.add('converse-muc-views', {
                 }
             },
 
+            destroy (groupchat, reason, onSuccess, onError) {
+                const destroy = $build("destroy");
+                const iq = $iq({to: groupchat, type: "set"}).c("query", {xmlns: Strophe.NS.MUC_OWNER}).cnode(destroy.node);
+                if (reason && reason.length > 0) { iq.c("reason", reason); }
+                return _converse.api.sendIQ(iq).then(onSuccess).catch(onError);
```

Comment:
I'd like us to move away from using callbacks in favour of using promises.

Here, you can simply return `return _converse.api.sendIQ(iq)` and the later when you call `destroy` you can either use `then` and `catch` or `async/await`.